### PR TITLE
TRUNK-5115-new-2:Replace use of conditional logging with slf4j parame…

### PR DESCRIPTION
**Pull request by gracebish**
…terized messages

**Pull request title** 
TRUNK-5115-new-2: Replace use of conditional logging with slf4j parameterized messages
**Jira Issue Title**
TRUNK-5115: Replace use of conditional logging with slf4j parameterized messages
**Description of what I changed**
Replace use of conditionals which are aimed at preventing to construct the debug messages with slf4j parameterized messages
**Issue I worked on**
https://issues.openmrs.org/browse/TRUNK-5115

## Checklist: I completed these to help reviewers :)

- [ x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by gracebish** [comment]
@sacull. Here are more changes for you to review

File: api/src/main/java/org/openmrs/module/ModuleClassLoader.java [hunk 1]
```diff
@@ -822,10 +812,7 @@ protected File cacheLibrary(final URL libUrl, final String libname) {
 			// save a link to the cached file
 			libraryCache.put(libUri, result);
 			
-			if (log.isDebugEnabled()) {
-				log.debug("library " + libname + " successfully cached from URL " + libUrl + " and saved to local file "
-				        + result);
-			}
+			log.debug("library {} successfully cached from URL {} and saved to local file {}", libname, libUrl, result);
```
Comment (sacull) [inline api/src/main/java/org/openmrs/module/ModuleClassLoader.java:815]: [Here](https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-loggingLogging) I can see that version 1.6.x has problems when there are more than two arguments. Have you checked if the currently used version still has this problem?

**Comment by HerbertYiga** [comment]
@gracebish   Well done on this grace,a better description on this could be some think like (TRUNK-5115: Replace use of conditional logging with slf4j parameterized messages),not sure why you have a new key word on your description

**Comment by gracebish** [comment]
I have rad through the document but failed to understand where exactly the
problem is. I thought the code has a problem if some tests fail to pass but
all checks seem to have passed. could you kindly point me to the exact
problem?

**Review by sacull — APPROVED** [review]
If everything works fine, that's fine for me. ;)

File: api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java [hunk 2]
```diff
@@ -590,26 +585,19 @@ protected static void nullInstance(Object instance) {
 					if (null != value) {
 						Class<?> valueClass = value.getClass();
 						if (!loadedByThisOrChild(valueClass)) {
-							if (log.isDebugEnabled()) {
-								log.debug("Not setting field " + field.getName() + " to null in object of class "
-										+ instance.getClass().getName() + " because the referenced object was of type "
-										+ valueClass.getName() + " which was not loaded by this WebappClassLoader.");
-							}
+							log.debug(
+							    "Not setting field {} to null in object of class {} because the referenced object was of type {} which was not loaded by this WebappClassLoader. {}",
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java:589]: Are you passing in 3 or 4 arguments?

Comment (gracebish) [reply api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java]: I have made the modifications on the ticket.

File: api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java [hunk 3]
```diff
@@ -590,26 +585,19 @@ protected static void nullInstance(Object instance) {
...
+							log.debug(
+							    "Not setting field {} to null in object of class {} because the referenced object was of type {} which was not loaded by this WebappClassLoader.",
+							    field.getName(), instance.getClass().getName(), valueClass.getName());
 						} else {
 							field.set(instance, null);
-							if (log.isDebugEnabled()) {
-								log.debug("Set field " + field.getName() + " to null in class "
-										+ instance.getClass().getName());
-							}
+							log.debug("Set field {} to null in class", field.getName(), instance.getClass().getName());
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java:593]: How many messages are you replacing in the above line?

File: api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java [hunk 4]
```diff
@@ -590,7 +590,7 @@ protected static void nullInstance(Object instance) {
 							    field.getName(), instance.getClass().getName(), valueClass.getName());
 						} else {
 							field.set(instance, null);
-							log.debug("Set field {} to null in class", field.getName(), instance.getClass().getName());
+							log.debug("Set field {} to null in class {}", field.getName(), instance.getClass().getName());
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java:593]: Was the comma before or after the field name?

Comment (gracebish) [reply api/src/main/java/org/openmrs/util/OpenmrsClassLoader.java:593]: The comma was after field.getName()

**Pull request merged**